### PR TITLE
changed checkbox to radio for mapping in target report filter, Fixes #679

### LINKED ISSUE #679: Change check box to radio as filter used in target output shows only one mapping
## Expected Behavior

Filter provided in the target output should shows radio button in mapping field.

## Current Behavior

Filter provided in the target output shows check box in mapping field. In spite of selecting multiple mapping it only shows one mapping which is right most in the list.

## Possible Solution

Change check box to radio button

## Steps to Reproduce (for bugs)

1. Run owtf and then switch to web ui
2. Select a target and click on filter option
3. Choose multiple mapping options

## Your Environment

* Installation method used:
* Version/branch used: develop
* Operating System and version (like Kali, Debian, ArchLinux, etc): Kali
* pip/setuptools version: 8.x
* sudo access: no

### PR DESCRIPTION
**Pull request by DarKnight24**
changed checkbox to radio for mapping in target report filter

## Description

Mapping only shows one mapping, hence radio buttons are suited for this filter

## Related Issue

#679 

## Reviewers

@delta24 

## Screenshots (if appropriate):

screenshot from 2016-05-05 15 58 51

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other

### Checklist:

- [x] My code follows the code style (modified PEP8) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

…#679

### CODE REVIEW

**Comment by delta24** [comment]
@DarKnight-- 1 thing needs fixing. If I revert back to the default OWTF mappings, then there is no way to do it with a radio butto. You would then need to clear **ALL** the filters just to deselect the radio button.

**Comment by DarKnight24** [comment]
@delta24 Updated the PR accordingly. Have a look at it.

**Comment by delta24** [comment]
@DarKnight24 update the PR to reflect the pip fix.

File: framework/interface/templates/target.html [hunk 1]
```diff
@@ -128,6 +143,13 @@ <h4 class="modal-title" id="pluginOutputFilterModalLabel">Advanced Filter</h4>
     }
 }
 
+function modifyMapping(radio, filter_type, data){
+    if (typeof mySpace.filter_data[filter_type] === 'undefined') mySpace.filter_data[filter_type] = [];
```
Comment (delta24) [inline framework/interface/templates/target.html]: Don't use inline if statements. Better to break it up and indent for easy reviewing.

**Comment by DarKnight24** [comment]
@delta24 Updated the PR

File: framework/interface/templates/target.html [hunk 2]
```diff
@@ -120,14 +135,22 @@ <h4 class="modal-title" id="pluginOutputFilterModalLabel">Advanced Filter</h4>
 }
 
 function modifyFilter(checkbox, filter_type, data){
-    if (typeof mySpace.filter_data[filter_type] === 'undefined') mySpace.filter_data[filter_type] = [];
+    if (typeof mySpace.filter_data[filter_type] === 'undefined')
```
Comment (delta24) [inline framework/interface/templates/target.html]: Also wrap the if conditionals in parenthesis like

```
if (typeof mySpace.filter_data[filter_type] === 'undefined'){
  mySpace.filter_data[filter_type] = [];
}
```

File: framework/interface/templates/target.html [hunk 3]
```diff
@@ -120,14 +135,22 @@ <h4 class="modal-title" id="pluginOutputFilterModalLabel">Advanced Filter</h4>
...
         mySpace.filter_data[filter_type].push(data);
     } else if ((!checkbox.checked) && (mySpace.filter_data[filter_type].indexOf(data) > -1)) {
         mySpace.filter_data[filter_type].splice(mySpace.filter_data[filter_type].indexOf(data), 1);
     }
 }
 
+function modifyMapping(radio, filter_type, data){
+    if (typeof mySpace.filter_data[filter_type] === 'undefined') 
+		mySpace.filter_data[filter_type] = [];
+    if (radio.checked)
```
Comment (delta24) [inline framework/interface/templates/target.html]: Same here.

**Pull request merged**